src/story/ant/ui.rs

```diff
@@ -570,38 +564,87 @@ pub fn on_added_ant_emote(
-                                    },
-                                    texture,
-                                    ..default()
-                                },
-                            ));
-                        });
-                }
-            }
+    for (ant_view_entity, emote, children) in ant_view_query.iter() {
+        if let Some(ant_sprite_view_entity) = children
+            .iter()
+            .find(|&&child| ant_sprite_view_query.get(child).is_ok())
+        {
+            commands
+                .entity(*ant_sprite_view_entity)
+                .with_children(|parent| {
+                    let texture = match emote.emote_type() {
+                        EmoteType::Asleep => asset_server.load("images/zzz.png"),
+                        EmoteType::FoodLove => asset_server.load("images/foodlove.png"),
+                    };
+
+                    parent.spawn((
+                        EmoteSprite {
+                            parent_entity: ant_view_entity,
+                        },
+                        SpriteBundle {
+                            transform: Transform::from_xyz(0.75, 1.0, 1.0),
+                            sprite: Sprite {
+                                custom_size: Some(Vec2::splat(1.0)),
+                                ..default()
+                            },
+                            texture,
+                            ..default()
+                        },
+                    ));
+                });
+        }
+    }
+}
+
+pub fn on_ant_ate_food(
+    mut ant_action_events: EventReader<AntAteFoodEvent>,
+    mut commands: Commands,
+    model_view_entity_map: Res<ModelViewEntityMap>,
+) {
+    for AntAteFoodEvent(ant_model_entity) in ant_action_events.read() {
+        let ant_view_entity = match model_view_entity_map.0.get(ant_model_entity) {
+            Some(ant_view_entity) => *ant_view_entity,
+            None => continue,
+        };
+
+        commands
+            .entity(ant_view_entity)
+            .insert(Emote::new(EmoteType::FoodLove));
+    }
+}
+
+pub fn ants_sleep_emote(
+    ants_query: Query<Entity, (With<Asleep>, With<AtNest>)>,
+    mut commands: Commands,
+    mut rng: ResMut<GlobalRng>,
+    settings: Res<Settings>,
+    model_view_entity_map: Res<ModelViewEntityMap>,
+) {
+    for ant_model_entity in ants_query.iter() {
+        if rng.f32() < settings.probabilities.sleep_emote {
+            let ant_view_entity = match model_view_entity_map.0.get(&ant_model_entity) {
+                Some(ant_view_entity) => *ant_view_entity,
+                None => continue,
+            };
+
+            commands
+                .entity(ant_view_entity)
+                .insert(Emote::new(EmoteType::Asleep));
+        }
+    }
+}
+
+pub fn on_ant_wake_up(
+    // TODO: Maybe this should be event-driven and have an AntWakeUpEvent instead? That way this won't run when ants are getting destroyed.
+    mut removed: RemovedComponents<Asleep>,
```

Comment:
As the comment mentions, I found it odd that there wasn't parity between eating emote and sleeping emote. It seems defensible to say that there should be `AntFellAsleep` and `AntWokeUp` events, but I didn't want to change more code than needed when the intent of this PR was to address the view/model concern specifically. So, I left it as-is.

Reply:
Have you tried just grouping the emote entity with ant entity, and controlling visibility instead of this?

Reply:
I'm not sure I understand what you're asking. Are you saying to always render the view and toggle the views visibility rather than conditionally adding/removing the view when the emote is required?

Reply:
Merging now but happy to talk more about this!